src/controllers/ContinuedIllnessController.ts

```diff
@@ -0,0 +1,77 @@
+import Joi from '@hapi/joi';
+import { SessionMiddleware } from 'ch-node-session-handler';
+import { controller } from 'inversify-express-utils';
+import { BaseController } from './BaseController';
+import { FormValidator } from './validators/FormValidator';
+
+import { AuthMiddleware } from 'app/middleware/AuthMiddleware';
+import { IllnessReasonFeatureMiddleware } from 'app/middleware/IllnessReasonFeatureMiddleware';
+import { loggerInstance } from 'app/middleware/Logger';
+import { Appeal } from 'app/models/Appeal';
+import { Illness } from 'app/models/Illness';
+import { YesNo } from 'app/models/fields/YesNo';
+import { createSchema } from 'app/models/fields/YesNo.schema';
+import { CONTINUED_ILLNESS_PAGE_URI } from 'app/utils/Paths';
+import { Navigation } from 'app/utils/navigation/navigation';
+
+const template = 'continued-illness';
+
+const navigation : Navigation = {
+    previous(): string {
+        return '';
+    },
+    next(): string {
+        return '';
+    },
+    actions: (_: boolean) => {
+        return {
+            continue:'action=continue'
+        };
+    }
+};
+
+@controller(CONTINUED_ILLNESS_PAGE_URI, IllnessReasonFeatureMiddleware, SessionMiddleware, AuthMiddleware)
+export class ContinuedIllnessController extends BaseController<any> {
+
+    constructor() {
+        const errorMessage = 'You must tell us if this is a continued illness';
+        const schema: Joi.AnySchema = Joi.object({
+            continuedIllness: createSchema(errorMessage)
+        }).unknown(true);
+
+        super(
+            template,
+            navigation,
+            new FormValidator(schema)
+        );
+    }
+
+    protected prepareViewModelFromAppeal(appeal: Appeal): any {
+        const illnessStartDateRaw = appeal.reasons.illness?.illnessStart;
+
+        if (!illnessStartDateRaw) {
+            throw new Error('Illness Start Date expected but not found');
+        } else {
+            const illnessStart = illnessStartDateRaw.toLocaleDateString('en-GB', {
+                year: 'numeric', month: 'long', day: 'numeric'
+            });
+            const hint: string = `You told us the illness started on ${illnessStart}`;
+            return { hint };
+        }
+    }
+
+    protected prepareSessionModelPriorSave(appeal: Appeal, value: Illness): Appeal {
+        if (appeal.reasons?.illness != null) {
+            appeal.reasons.illness.continuedIllness = value.continuedIllness;
+            if (value.continuedIllness === YesNo.yes) {
```

Comment:
Feels unnecessary as values are undefined by default.

Reply:
True, however this logic covers the following scenario:

1.  User enters illness journey and ticks "Is the person still ill?" as No
2.  User is then asked to enter an illness end date
3.  User gets near the end of the journey, decides they've made a mistake, and navigates back to this page
4.  User then instead ticks Yes for "Is the person still ill"

In this scenario, without resetting `illnessEnd` to `undefined`, the illness would have conflicting values:
* `continuedIllness`: Yes
* `illnessEnd`: <Some date in the past>

This wouldn't make sense from a business perspective, so we must nullify `illnessEnd` whenever `Yes` is chosen on this page.

Reply:
Good call, forgot about it. Ignore my previous comment in that case.